conanfile.py

```diff
@@ -107,15 +123,22 @@ def source(self):
             submodules.append("qtwebsockets")
         if self.options.xmlpatterns:
             submodules.append("qtxmlpatterns")
+        if self.options.declarative:
+            submodules.append("qtdeclarative")
+        if self.options.multimedia:
+            submodules.append("qtmultimedia")
+        if self.options.doc:
+            submodules.append("qtdoc")
+        if self.options.repotools:
+            submodules.append("qtrepotools")
+        if self.options.qa:
+            submodules.append("qtqa")
+        if self.options.quickcontrols:
+            submodules.append("qtquickcontrols")
+
+        if len(submodules) > 0:
+            self.run("cd %s && git submodule update --init %s" % (self.source_dir, " ".join(submodules)))
```

Comment:
It seem to me that getting sources during the build step is not a good thing. I have a working solution getting all sources during the source step, using qt-everywhere-opensource-src archive from qt.io, which is the recommended way to build qt.

Reply:
Totally agree, thanks a lot for this solution! can you make the PR as discussed 😊